lib/intercom-rails/auto_include_filter.rb

```diff
@@ -56,7 +65,16 @@ def intercom_script_tag_called_manually?
       end
 
       def intercom_script_tag
-        @script_tag ||= ScriptTag.new(:find_current_user_details => true, :find_current_company_details => true, :controller => controller, :show_everywhere => show_everywhere?)
+        # User defined method for applying a nonce to the inserted js tag when
+        # using CSP
+        if defined?(CoreExtensions::IntercomRails::AutoInclude.csp_nonce_hook) == 'method'
```

Comment:
I think this reads much clearer:
```ruby
options = {
  :find_current_user_details => true,
  :find_current_company_details => true,
  :controller => controller,
  :show_everywhere => show_everywhere?
}

if defined?(CoreExtensions::IntercomRails::AutoInclude.csp_nonce_hook) == 'method'
  options.merge!(:nonce => CoreExtensions::IntercomRails::AutoInclude.csp_nonce_hook(controller.request))
end

@script_tag = ScriptTag.new(options)
```

What do you think?

I don't think the memoization (`||=-`) is necessary. If it is then a spec will fail when we change